lib/cocoapods/generator/embed_frameworks_script.rb

```diff
@@ -71,6 +71,14 @@ def script
 # Copies and strips a vendored framework
 install_framework()
 {
+  local basename
+  basename="$(basename -s .framework "$1")"
+  source_binary="$1/${basename}"
+  if [[ $(file "$source_binary" | grep 'current ar archive') ]]; then
```

Comment:
We don't need to wait until this script runs to do this check - we know at the time this script is created during `pod install` and can skip it entirely. We had logic for this but it must have gotten lost during some refactoring.

Here's the relevant code that could be updated to fix this:

https://github.com/CocoaPods/CocoaPods/blob/11b34726a83067ecc66b364e2c5d620fd303513e/lib/cocoapods/generator/embed_frameworks_script.rb#L160-L165

changing this portion: 
```ruby
xcframeworks.each do |xcframeworks|
```

to be this:

```ruby
xcframeworks.select { |xcf| xcf.build_type.dynamic_framework? }.each do |xcframework|
```

should do the trick. That will remove the call to `install_framework` entirely

Reply:
awesome, if we do that @BernardGatt can you add a test also?

Reply:
Agreed, makes more sense.

@dnkoutso working on a test.